merino/providers/weather/backends/accuweather.py

```diff
@@ -92,6 +110,87 @@ def cache_inputs_for_weather_report(self, geolocation: Location) -> Optional[byt
 
         return (geolocation.country + geolocation.postal_code).encode("utf-8")
 
+    def cache_key_for_accuweather_request(
+        self, url: str, query_params: dict[str, str] = {}
+    ) -> str:
+        """Get the cache key for the accuweather request.
+        Also ensure that the API key is stripped out.
+        """
+        params = query_params.copy()
+        if "apikey" in params:
+            del params["apikey"]
+
+        if params:
+            extra_identifiers = hashlib.blake2s(str(params).encode("utf-8")).hexdigest()
```

Comment:
```suggestion
            hasher = hashlib.blake2s()
            for key, value in sorted(params.items()):
                hasher.update(key.encode("utf-8") + value.encode("utf-8"))
            extra_identifiers = hasher.hexdigest()
```

suggestion: I'm not sure if the dictionary string representation—or dictionary enumeration order—is guaranteed to be stable, so we might end up with spurious cache misses for the same set of `params`, depending on how exactly the dictionary is populated.

It's a bit more code, but for cases like this, I like to explicitly canonicalize the representation.

Reply:
Oh good point! I think Python 3.7+ [has stable `dict` enumeration order](https://mail.python.org/pipermail/python-dev/2017-December/151283.html) (which makes me presume that the string will be stable too). However, `dicts` should _not_ be assumed to be ordered, as that's not the intent of this type of data structure. So let's make this explicit!

Reply:
On top of Lina's suggestion, we can do this to save a dict copy and a `dict.pop()` for all current condition and forecast requests.
```suggestion
           if len(params) > 1:  // "apikey" is included for all requests
                hasher = hashlib.blake2s()
                for key, value in sorted(params.items()):
                    if key != self.url_param_api_key:
                        hasher.update(key.encode("utf-8") + value.encode("utf-8"))
                extra_identifiers = hasher.hexdigest()
```

Reply:
> we can do this to save a dict copy and a dict.pop() for all current condition and forecast requests.

Oh! This is actually a much better idea. I'll do this instead, thanks!